### DISCUSSION TITLE
How to remove Source/Layer from Map?

### DISCUSSION THREAD

**Discussion by nonoumasy** [uber/react-map-gl#1854] [Q&A] [answered]
I'm trying to dynamically add GeoJson to the map. I can add each item using the <Source/> and <Layer/> combo. However, once its added to the map, I can't seem to be able to remove it completely. It will remain semi-opaque.

I made a prototype here:
https://codesandbox.io/s/mapbox-with-geojson-egxb6?file=/src/components/MapComp.js

<img width="1440" alt="Screen Shot 2022-05-10 at 8 56 30 PM" src="https://user-images.githubusercontent.com/6312676/167766135-0b14c622-d6a1-41c8-a4e9-2d018a616671.png">

**Comment by Pessimistress** [comment, marked as answer]
You need to add [key props](https://reactjs.org/docs/lists-and-keys.html#keys) to your Source components if you are arbitrarily adding and removing them in a list.
